### PR TITLE
Refactor queue_options in QueueConfig to be Tuple[str, str]

### LINKED ISSUE #6149: Refactor queue_options to be Tuple[str,str]
Currently queue_options are a Union of Tuple[str,str] and str, where the latter part is used solely for unsetting options.
This however is handled using empty strings later on, which we could have defaulted to using in the first place.

We should be able to simplify some code by reducing the Union to the Tuple only.

queue_config.py
```
        queue_options: Dict[
            QueueSystem, List[Union[Tuple[str, str], str]]
        ] = defaultdict(list)
        for system, option_name, *values in config_dict.get("QUEUE_OPTION", []):
            queue_system = QueueSystem.from_string(system)
            if option_name not in VALID_QUEUE_OPTIONS[queue_system]:
                raise ConfigValidationError(
                    f"Invalid QUEUE_OPTION for {queue_system.name}: '{option_name}'. "
                    f"Valid choices are {sorted(VALID_QUEUE_OPTIONS[queue_system])}."
                )
            if values:
                queue_options[queue_system].append((option_name, values[0]))
            else:
                queue_options[queue_system].append(option_name)
```

driver.py
```
def create_driver(cls, queue_config: QueueConfig) -> "Driver":
    driver = Driver(queue_config.queue_system)
    if queue_config.queue_system in queue_config.queue_options:
        for setting in queue_config.queue_options[queue_config.queue_system]:
            if isinstance(setting, tuple):
                driver.set_option(*setting)
            else:
                driver.unset_option(setting)
    return driver


def unset_option(self, option: str) -> None:
	if option == "MAX_RUNNING":
		self.set_max_running(0)
	else:
		self._set_option(option, “”)
```

### PR DESCRIPTION
**Pull request by larsevj**
**Issue**
Resolves #6149 

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unit_tests/job_queue/test_driver.py [hunk 1]
```diff
@@ -12,15 +12,15 @@ def test_set_and_unset_option():
...
+                ("MAX_RUNNING", ""),
             ]
         },
     )
     driver = Driver.create_driver(queue_config)
     assert driver.get_option("MAX_RUNNING") == "0"
     assert driver.set_option("MAX_RUNNING", "42")
     assert driver.get_option("MAX_RUNNING") == "42"
-    driver.unset_option("MAX_RUNNING")
+    driver.set_option("MAX_RUNNING", "")
```
Comment (andreas-el) [inline tests/unit_tests/job_queue/test_driver.py:23]: Do you also test the other way of setting this?

E.g.
```
driver.set_option("MAX_RUNNING", "0")
assert driver.get_option("MAX_RUNNING") == "0"
```

File: src/ert/config/queue_config.py [hunk 2]
```diff
@@ -93,8 +91,7 @@ def from_dict(cls, config_dict: ConfigDict) -> QueueConfig:
             if values:
                 queue_options[queue_system].append((option_name, values[0]))
             else:
-                queue_options[queue_system].append(option_name)
-
+                queue_options[queue_system].append((option_name, ""))
```
Comment (andreas-el) [inline src/ert/config/queue_config.py:94]: Could this work?
```
queue_options[queue_system].append((option_name, values[0] if values else ""))
```

File: src/ert/config/queue_config.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def __post_init__(self) -> None:
             setting
             for settings in self.queue_options.values()
             for setting in settings
-            if isinstance(setting, tuple) and setting[0] == "MAX_RUNNING"
+            if setting[0] == "MAX_RUNNING" and setting[1]
```
Comment (andreas-el) [inline src/ert/config/queue_config.py:40]: Will probably work, but do we need that extra for-loop here after the union was removed?
Also this seems a bit excessive way to test this one option, can this be refactored somehow?

**Review by andreas-el — CHANGES_REQUESTED** [review]
Almost there, great work this! 👍

File: src/ert/config/queue_config.py (hunk 3 above)
Comment (larsevj) [reply src/ert/config/queue_config.py]: I put this check into the main for-loop, but unsure if this improves anything. The validations are all over the place in this class.

Comment (larsevj) [reply src/ert/config/queue_config.py:40]: I believe that with the current structure of `queue_options`, the extra for loop is needed to unpack all values correctly from .values().

File: tests/unit_tests/config/test_queue_config.py [hunk 4]
```diff
@@ -185,3 +184,23 @@ def test_undefined_LSF_SERVER_environment_variable():
...
+        f.write("NUM_REALIZATIONS 1\n")
+        f.write(f"QUEUE_SYSTEM {queue_system}\n")
+        f.write(f"QUEUE_OPTION {queue_system} {queue_system_option}\n")
+        f.write(f"QUEUE_OPTION {queue_system} MAX_RUNNING\n")
+    config_object = ErtConfig.from_file(filename)
+    driver = Driver.create_driver(config_object.queue_config)
+    assert driver.get_option(queue_system_option) == ""
+    assert driver.get_option("MAX_RUNNING") == "0"
+    for options in config_object.queue_config.queue_options[queue_system]:
+        assert isinstance(options, tuple)
```
Comment (andreas-el) [inline tests/unit_tests/config/test_queue_config.py:206]: Is this leftovers, or can options be something else than a `Tuple[str,str]` ?

**Review by andreas-el — APPROVED** [review]
Looks good! 🚀

File: tests/unit_tests/config/test_queue_config.py (hunk 4 above)
Comment (larsevj) [reply tests/unit_tests/config/test_queue_config.py:206]: Just there to verify that every option is now a tuple when giving in empty values.

**Pull request merged**